pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go

```diff
@@ -199,6 +199,13 @@ type CiliumBGPVirtualRouter struct {
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default=false
 	ExportPodCIDR *bool `json:"exportPodCIDR,omitempty"`
+	// PodIPPoolSelector selects CiliumPodIPPools based on labels. The virtual
+	// router will announce allocated CIDRs of matching CiliumPodIPPools.
+	//
+	// If empty / nil no CiliumPodIPPools will be announced.
```

Comment:
Hm. How does one announce all pools? Does this require that every pool has a label?

Given that the service selector also has this behavior, it's probably a sensible default, but I'm still curious how to do that. Because I expect that it will be very common that users want to announce all pools that a node has pod IPs allocated from.

Reply:
Yes, the `PodIPPoolSelector` is consistent with the service selector by only selecting pools with matching labels (and allocated to the local CiliumNode).

> I expect that it will be very common that users want to announce all pools that a node has pod IPs allocated from.

We can reshape the API to expose other selection options, e.g. all pools from the cluster or a particular namespace. Since this is already a large PR and these are alpha API changes, are you OK with supporting this use case in a follow-on PR? If so, I can create an issue and assign myself.

Reply:
In K8s, typically an empty selector means it selects all. A nil selector selects nothing. At least the former is how Cilium's CNP `endpointSelector` works. The reason the latter is not implemented is because CNPs cannot have a nil `endpointSelector`. 

But 

* empty => all
* nil => nil

makes sense to me. Thoughts @cilium/sig-k8s?

Reply:
Given that the CRD is alpha, we don't have to be extra careful about breaking the API, so this is ok with me for now. However, I think we need to have a larger discussion on how these selectors behave differently across resources and wider compared to K8s. Could you bring this up during the next community meeting?